# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsGridContainerFrame.h
# Commit: 2ac943ae4424
# Full Hash: 2ac943ae442445f96f5a68d5298b320f30b1313a
# Author: Mats Palmgren <mats@mozilla.com>
# Date: 2019-10-13 09:36:01
# Regressor Bug: 1234311
# File Overlap Count: 1
# Description:
#   Bug 1234311 part 2 - [css-grid-2] Implement 'grid-template-rows/columns' resolved value correctly for subgrid.  r=dholbert
#   
#   This implements the resolution and adds WPTs for:
#   https://github.com/w3c/csswg-drafts/issues/4362
#   
# ==============================================================================

diff -r 43f18d905d75 -r 2ac943ae4424 layout/generic/nsGridContainerFrame.h
--- a/layout/generic/nsGridContainerFrame.h	Sat Oct 12 17:15:43 2019 +0000
+++ b/layout/generic/nsGridContainerFrame.h	Sat Oct 12 17:16:02 2019 +0000
@@ -45,7 +45,8 @@
       nsTArray<nscoord>&& aPositions, nsTArray<nscoord>&& aSizes,
       nsTArray<uint32_t>&& aStates, nsTArray<bool>&& aRemovedRepeatTracks,
       uint32_t aRepeatFirstTrack,
-      nsTArray<nsTArray<StyleCustomIdent>>&& aResolvedLineNames)
+      nsTArray<nsTArray<StyleCustomIdent>>&& aResolvedLineNames,
+      bool aIsSubgrid)
       : mNumLeadingImplicitTracks(aNumLeadingImplicitTracks),
         mNumExplicitTracks(aNumExplicitTracks),
         mStartFragmentTrack(aStartFragmentTrack),
@@ -54,8 +55,9 @@
         mSizes(aSizes),
         mStates(aStates),
         mRemovedRepeatTracks(aRemovedRepeatTracks),
+        mResolvedLineNames(std::move(aResolvedLineNames)),
         mRepeatFirstTrack(aRepeatFirstTrack),
-        mResolvedLineNames(std::move(aResolvedLineNames)) {}
+        mIsSubgrid(aIsSubgrid) {}
   uint32_t mNumLeadingImplicitTracks;
   uint32_t mNumExplicitTracks;
   uint32_t mStartFragmentTrack;
@@ -64,8 +66,9 @@
   nsTArray<nscoord> mSizes;
   nsTArray<uint32_t> mStates;
   nsTArray<bool> mRemovedRepeatTracks;
+  nsTArray<nsTArray<StyleCustomIdent>> mResolvedLineNames;
   uint32_t mRepeatFirstTrack;
-  nsTArray<nsTArray<StyleCustomIdent>> mResolvedLineNames;
+  bool mIsSubgrid;
 };
 
 struct ComputedGridLineInfo {